sys/include/net/dns.h

```diff
@@ -74,6 +74,15 @@ static inline int dns_query(const char *domain_name, void *addr_out, int family)
 {
     int res = -ENOTSUP;
 
+    if (family == AF_UNSPEC) {
+        if (!IS_USED(MODULE_IPV4_ADDR)) {
+            family = AF_INET6;
+        }
+        else if (!IS_USED(MODULE_IPV6_ADDR)) {
+            family = AF_INET;
+        }
```

Comment:
Can this be written like this:

```suggestion
        if (IS_USED(MODULE_IPV6_ADDR)) {
            family = AF_INET6;
        }
        else if (IS_USED(MODULE_IPV4_ADDR)) {
            family = AF_INET;
        }
```

That seems much more readable but maybe there's something I'm missing.

Reply:
If both IPv4 and IPv6 are enabled, `AF_UNSPEC` can stay as the stack can deal with both A and AAAA responses.